### PR TITLE
[LWDM] refactor(noble-curves): replace APIs 2.x renames away (LIVE-37378)

### PR DESCRIPTION
**Pull request by gre-ledger**
### 📝 Description

`@noble/curves` 2.x removes nine APIs the tree still uses. Every replacement already exists on the pinned 1.9.7, so the rename lands ahead of the bump rather than inside it.

| Removed in 2.x | Replacement |
| --- | --- |
| `secp256k1.ProjectivePoint` | `secp256k1.Point` |
| `point.toRawBytes(c)` | `point.toBytes(c)` |
| `Point.fromHex(<bytes>)` (narrows to string-only) | `Point.fromBytes(<bytes>)` |
| `Point.fromPrivateKey(pk)` | `Point.BASE.multiply(Point.Fn.fromBytes(pk))` |
| `secp256k1.CURVE.n` | `secp256k1.Point.CURVE().n` |
| `utils.isValidPrivateKey(k)` | `utils.isValidSecretKey(k)` |
| `Signature.fromDER(x)` | `Signature.fromBytes(x, "der")` |
| `Signature.fromCompact(x)` | `Signature.fromBytes(x, "compact")` |
| `sig.toCompactRawBytes()` / `sig.toDERRawBytes()` | `sig.toBytes("compact")` / `sig.toBytes("der")` |

Each substitution was checked byte-for-byte against 1.9.7 before applying, including that `Point.BASE.multiply(Point.Fn.fromBytes(pk))` still rejects `0`, `n` and `n+1` exactly like the removed `fromPrivateKey` — the `isPrivate` wrapper in `BtcNew.test.ts` depends on that throw.

One non-rename cleanup: `wallet-btc`'s `toRawBytes(point.toRawBytes().length === 33)` allocated a 33-byte array only to compare its length against a condition that is always true (the default is compressed on both versions). It is now `toBytes(true)` — identical output, one less allocation on a path that runs per address during sync.

**Coverage.** Most migrated call sites were already covered — the `wallet-btc` taproot path by 200 exact bech32m vectors, the `hw-app-btc` taproot output key by the `p2tr` specs, the swap DER conversion by its leading-zero R/S edge cases, and the key-ring ECC wrapper by bip32 v4's `testEcc()`, which runs known-answer vectors at `BIP32Factory()` module load. Two had none, so this adds three tests, both mutation-verified to fail when the migration is wrong:

- `buildKaspaXpub` — round-tripped through coin-kaspa's own `parseExtendedPublicKey`, plus an exact 99-byte assertion. Fixture is the SEC2 generator point, so the ground truth does not come from a curve library.
- key-ring `ecdh` — a known-answer vector from OpenSSL (`node:crypto` `createECDH`). The existing test only checked that two parties agree, which passes even when both sides are wrong in the same way.

> [!NOTE]
> `sign()` is deliberately **not** migrated here. It returns a `Signature` on 1.9.7 and a plain `Uint8Array` on 2.x — a type change, not a rename — so no single expression works on both versions (`format: "compact"` still returns a `Signature` on 1.9.7). The 8 affected call sites must move with the bump, tracked in LIVE-37678. The sharpest one is `coin-tester-tron`, where `out[64] = sig.recovery ?? 0` degrades silently after the bump: a well-formed 65-byte signature that recovers the wrong address.

### 🔗 Context

- **JIRA / GitHub issue**: [LIVE-37378](https://ledgerhq.atlassian.net/browse/LIVE-37378) — follow-up [LIVE-37678](https://ledgerhq.atlassian.net/browse/LIVE-37678)
- **ADR** (if any): n/a

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[LIVE-37378]: https://ledgerhq.atlassian.net/browse/LIVE-37378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by gre-ledger** [comment]
## 🧪 Triggered test workflows

Manually dispatched on `refactor/noble-curves-LIVE-37378-migrate-projectivepoint` (commit 6e293cb38b9), **scoped to the diff** rather than the full suite:

| Workflow | Scope | Run |
|---|---|---|
| [Desktop] E2E Only | `@bitcoin,@family-bitcoin,@zcash,@family-zcash,Ledger.Sync` — Speculos nanoSP | [run 35235574860](https://github.com/LedgerHQ/ledger-live/actions/runs/35235574860) |
| [Mobile] E2E Only | `@bitcoin,@family-bitcoin,ledgerSync` — Android only, nanoSP | [run 35235579062](https://github.com/LedgerHQ/ledger-live/actions/runs/35235579062) |

Coin Tester is **not** dispatched — it auto-runs on this PR and already covers every affected chain (kaspa, tron, bitcoin, tezos, zcash), all of whose `src/signer.ts` this PR touches.

Why these three surfaces:

- **Bitcoin** — `wallet-btc/src/crypto/secp256k1.ts` (`publicKeyTweakAdd`) and `hw-app-btc`'s `bip32.ts` / `BtcNew.ts` / `newops/accounttype.ts` drive xpub derivation and the taproot output key, so address derivation, receive, and send are the paths that would break. Desktop matters most here: it runs the noble default through the `publicKeyTweakAdd` web workers, whereas mobile overrides it with `react-native-fast-crypto` — so the mobile run is there for the shared `hw-app-btc` code, not for `wallet-btc`.
- **Zcash** — the zcash xpub compose path exercises the same uncompressed-point handling that changed.
- **Ledger Sync** — `hw-ledger-key-ring-protocol/src/NobleCrypto.ts` is the LKRP crypto (ECDH, BIP32 derive, sign/verify).

Android-only on mobile because nothing in the diff is platform-native.

File: libs/coin-tester-modules/coin-tester-bitcoin/src/signer.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ const eccWrapper = {
     const hashBytes = hash instanceof Buffer ? new Uint8Array(hash) : hash;
     const keyBytes = privateKey instanceof Buffer ? new Uint8Array(privateKey) : privateKey;
     const signature = secp256k1.sign(hashBytes, keyBytes, { prehash: false });
-    return signature.toCompactRawBytes();
+    return signature.toBytes("compact");
```
[the remark this answers is not in the document]
Comment (gre-ledger) [reply libs/coin-tester-modules/coin-tester-bitcoin/src/signer.ts:125]: Correct that `.toBytes("compact")` breaks on 2.x, but reverting to `toCompactRawBytes()` does not help: in 2.x `sign()` returns a plain `Uint8Array`, so **both** methods are `undefined`.

```
1.9.7   sign() -> Signature   | .toBytes: function  | .toCompactRawBytes: function
2.3.0   sign() -> Uint8Array  | .toBytes: undefined | .toCompactRawBytes: undefined
```

The old spelling is in fact doubly dead on 2.x — removed from `Signature` *and* no longer reachable, since no `Signature` is returned. So this line is exactly as 2.x-blocked either way, and keeping the deprecated name would just leave a second rename for LIVE-37678 to do. Migrating the name now and fixing the return shape there is the smaller total diff.

Resolving — the return-shape work is tracked in LIVE-37678.

File: libs/coin-tester-modules/coin-tester-tezos/src/signer.ts [hunk 2]
```diff
@@ -126,7 +126,7 @@ export async function buildTz2Signer(): Promise<TezosTestSigner> {
       const txBytes = Buffer.from(rawTxHex.replace(/^0x/, ""), "hex");
       const hash = blake2b(txBytes, { dkLen: 32 });
       const sig = secp256k1.sign(hash, privateKey);
-      return Buffer.from(sig.toCompactRawBytes()).toString("hex");
+      return Buffer.from(sig.toBytes("compact")).toString("hex");
```
[the remark this answers is not in the document]
Comment (gre-ledger) [reply libs/coin-tester-modules/coin-tester-tezos/src/signer.ts:129]: Correct that `.toBytes("compact")` breaks on 2.x, but reverting to `toCompactRawBytes()` does not help: in 2.x `sign()` returns a plain `Uint8Array`, so **both** methods are `undefined`.

```
1.9.7   sign() -> Signature   | .toBytes: function  | .toCompactRawBytes: function
2.3.0   sign() -> Uint8Array  | .toBytes: undefined | .toCompactRawBytes: undefined
```

The old spelling is in fact doubly dead on 2.x — removed from `Signature` *and* no longer reachable, since no `Signature` is returned. So this line is exactly as 2.x-blocked either way, and keeping the deprecated name would just leave a second rename for LIVE-37678 to do. Migrating the name now and fixing the return shape there is the smaller total diff.

Resolving — the return-shape work is tracked in LIVE-37678.

File: libs/hw-ledger-key-ring-protocol/src/NobleCrypto.ts [hunk 3]
```diff
@@ -120,7 +120,7 @@ const eccWrapper = {
     const hashBytes = hash instanceof Buffer ? new Uint8Array(hash) : hash;
     const keyBytes = privateKey instanceof Buffer ? new Uint8Array(privateKey) : privateKey;
     const signature = secp256k1.sign(hashBytes, keyBytes, { prehash: false });
-    return signature.toCompactRawBytes();
+    return signature.toBytes("compact");
```
[the remark this answers is not in the document]
Comment (gre-ledger) [reply libs/hw-ledger-key-ring-protocol/src/NobleCrypto.ts:123]: Correct that `.toBytes("compact")` breaks on 2.x, but reverting to `toCompactRawBytes()` does not help: in 2.x `sign()` returns a plain `Uint8Array`, so **both** methods are `undefined`.

```
1.9.7   sign() -> Signature   | .toBytes: function  | .toCompactRawBytes: function
2.3.0   sign() -> Uint8Array  | .toBytes: undefined | .toCompactRawBytes: undefined
```

The old spelling is in fact doubly dead on 2.x — removed from `Signature` *and* no longer reachable, since no `Signature` is returned. So this line is exactly as 2.x-blocked either way, and keeping the deprecated name would just leave a second rename for LIVE-37678 to do. Migrating the name now and fixing the return shape there is the smaller total diff.

Resolving — the return-shape work is tracked in LIVE-37678.

File: libs/ledger-live-common/src/exchange/swap/completeExchange.test.ts [hunk 4]
```diff
@@ -120,7 +120,7 @@ describe("enrichSwapSignatureVerificationError", () => {
   // must not hash it again (guards against a double-hash if the @noble/curves default ever changes).
   const signCompact = (curve: NobleCurve, message: Buffer): Buffer =>
     Buffer.from(
-      curve.sign(sha256(message), PRIVATE_KEY, { lowS: false, prehash: false }).toCompactRawBytes(),
+      curve.sign(sha256(message), PRIVATE_KEY, { lowS: false, prehash: false }).toBytes("compact"),
```
[the remark this answers is not in the document]
Comment (gre-ledger) [reply libs/ledger-live-common/src/exchange/swap/completeExchange.test.ts:123]: Correct that `.toBytes("compact")` breaks on 2.x, but reverting to `toCompactRawBytes()` does not help: in 2.x `sign()` returns a plain `Uint8Array`, so **both** methods are `undefined`.

```
1.9.7   sign() -> Signature   | .toBytes: function  | .toCompactRawBytes: function
2.3.0   sign() -> Uint8Array  | .toBytes: undefined | .toCompactRawBytes: undefined
```

The old spelling is in fact doubly dead on 2.x — removed from `Signature` *and* no longer reachable, since no `Signature` is returned. So this line is exactly as 2.x-blocked either way, and keeping the deprecated name would just leave a second rename for LIVE-37678 to do. Migrating the name now and fixing the return shape there is the smaller total diff.

Resolving — the return-shape work is tracked in LIVE-37678.

File: libs/ledgerjs/packages/hw-app-exchange/src/Exchange.integ.test.ts [hunk 5]
```diff
@@ -632,15 +632,15 @@ async function signMessage(
   const hash = new Uint8Array(hashBuffer);
 
   const signature = secp256k1.sign(hash, privKey, { prehash: false });
-  const sig = signature.toCompactRawBytes();
+  const sig = signature.toBytes("compact");
```
[the remark this answers is not in the document]
Comment (gre-ledger) [reply libs/ledgerjs/packages/hw-app-exchange/src/Exchange.integ.test.ts:635]: Correct that `.toBytes("compact")` breaks on 2.x, but reverting to `toCompactRawBytes()` does not help: in 2.x `sign()` returns a plain `Uint8Array`, so **both** methods are `undefined`.

```
1.9.7   sign() -> Signature   | .toBytes: function  | .toCompactRawBytes: function
2.3.0   sign() -> Uint8Array  | .toBytes: undefined | .toCompactRawBytes: undefined
```

The old spelling is in fact doubly dead on 2.x — removed from `Signature` *and* no longer reachable, since no `Signature` is returned. So this line is exactly as 2.x-blocked either way, and keeping the deprecated name would just leave a second rename for LIVE-37678 to do. Migrating the name now and fixing the return shape there is the smaller total diff.

Resolving — the return-shape work is tracked in LIVE-37678.

File: libs/coin-tester-modules/coin-tester-tron/src/signer.ts [hunk 6]
```diff
@@ -27,7 +27,7 @@ export function buildTronTestSignerFromPrivateKeyHex(privateKeyHex: string): {
     async sign(_path, rawTxHex, _tokenSignatures) {
       const sig = secp256k1.sign(sha256(Buffer.from(rawTxHex, "hex")), priv, { lowS: true });
       const out = new Uint8Array(65);
-      out.set(sig.toCompactRawBytes(), 0);
+      out.set(sig.toBytes("compact"), 0);
```
[the remark this answers is not in the document]
Comment (gre-ledger) [reply libs/coin-tester-modules/coin-tester-tron/src/signer.ts:30]: Same as the sibling threads: reverting to `toCompactRawBytes()` is not safer, because 2.x `sign()` returns a plain `Uint8Array` and both methods are `undefined` there.

On the second suggestion — branching on the two return shapes — that would work, but it means a runtime `instanceof Uint8Array` shim in every one of the 8 sign call sites, added now and deleted again at the bump. LIVE-37678 does it once, in the same change that moves the catalog.

Worth recording that this site is the sharpest of the 8, and for a reason neither spelling addresses: line 31 is `out[64] = sig.recovery ?? 0`. On 2.x `sig.recovery` is `undefined`, so the `?? 0` swallows it and writes recovery id 0 unconditionally — a well-formed 65-byte signature from which Tron recovers the wrong address most of the time, rather than a crash. That needs `format: "recovered"`, and it is called out explicitly in LIVE-37678.

Resolving.

**Review by CremaFR — APPROVED** [review]

**Review by francois-guerin-ledger — APPROVED** [review]

**Review by ysitbon — APPROVED** [review]

**Review by hedi-edelbloute — APPROVED** [review]
